addons/fleet/views/fleet_vehicle_cost_views.xml

```diff
@@ -334,6 +335,11 @@
         <field name="model">fleet.vehicle.log.contract</field>
         <field name="arch" type="xml">
             <search string="Vehicles Contracts">
+                <field name="vehicle_id" string="Vehicle" filter_domain="[('vehicle_id.name','ilike', self)]"/>
+                <field name="purchaser_id" string="Driver" filter_domain="[('purchaser_id','child_of', self)]"/>
```

Comment:
Actually after checking, as we link to a partner we don't have to make specific stuff about email. Remove all the emails !